### PR TITLE
Integrate with AWS cloud to appropriately tag control-plane instances

### PR DESCRIPTION
**Pull request by addyess**
Adds cloud_integration.py to prepare for all clouds Charmed-Kubernetes must relate to -- starting with AWS

Needs this PR first:
* https://github.com/juju-solutions/interface-aws-integration/pull/17

### CODE REVIEW

**Comment by addyess** [comment]
Addresses [LP#2055337](https://bugs.launchpad.net/charm-kubernetes-master/+bug/2055337)

**Review by eaudetcobello — COMMENTED** [review]
Another

File: src/cloud_integration.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            )
+            return
+
+        if not cloud.relation:
+            log.info(
+                "Skipping Cloud integration: Needs an active %s relation to integrate.", cloud_name
+            )
+            return
+
+        status.add(ops.MaintenanceStatus(f"Integrate with {cloud}"))
```
[the remark this answers is not in the document]
Comment (addyess) [reply src/cloud_integration.py:55]: wow great catch

**Review by eaudetcobello — APPROVED** [review]

File: metadata.yaml [hunk 2]
```diff
@@ -42,9 +42,6 @@ provides:
 requires:
   aws:
     interface: aws-integration
-    description: |
-      [Deprecated] only necessary for upgrades from 1.28 and can be removed after the upgrade
-      There's no need to relate if this is a fresh deployment
```
Comment (kwmonroe) [inline metadata.yaml:47]: bummer :(

**Review by kwmonroe — APPROVED** [review]
nit in the docstring, but otherwise lgtm

**Pull request merged**